### PR TITLE
update team_tag id format, fixes tf import of this resource

### PR DESCRIPTION
**Pull request by davidbloss**
## Changelog

- [X] List your changes here
- [X] Make a `changie` entry

### With this config, and empty Terraform state file
```tf
resource "opslevel_team_tag" "team_tag_1" {
  key        = "key2"
  value      = "value"
  team       = "Z2lkOi8vb3BzbGV2ZWwvVGVhbS8xNzQzNA"
  team_alias = "team_platform_3_a"
}
```

### Import `opslevel_team_tag` instance
```bash
terraform import opslevel_team_tag.team_tag_1 'Z2lkOi8vb3BzbGV2ZWwvVGVhbS8xNzQzNA:Z2lkOi8vb3BzbGV2ZWwvVGFnLzgyNjU0OTMw'
```
outputs
```tf
opslevel_team_tag.team_tag_1: Importing from ID "Z2lkOi8vb3BzbGV2ZWwvVGVhbS8xNzQzNA:Z2lkOi8vb3BzbGV2ZWwvVGFnLzgyNjU0OTMw"...
opslevel_team_tag.team_tag_1: Import prepared!
  Prepared opslevel_team_tag for import
opslevel_team_tag.team_tag_1: Refreshing state... [id=Z2lkOi8vb3BzbGV2ZWwvVGVhbS8xNzQzNA:Z2lkOi8vb3BzbGV2ZWwvVGFnLzgyNjU0OTMw]

Import successful!

The resources that were imported are shown above. These resources are now in
your Terraform state and will henceforth be managed by Terraform.
```

### Destroy imported `opslevel_team_tag`, `terraform destroy`
```tf
opslevel_team_tag.team_tag_1: Refreshing state... [id=Z2lkOi8vb3BzbGV2ZWwvVGVhbS8xNzQzNA:Z2lkOi8vb3BzbGV2ZWwvVGFnLzgyNjU0OTMw]

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  - destroy

Terraform will perform the following actions:

  # opslevel_team_tag.team_tag_1 will be destroyed
  - resource "opslevel_team_tag" "team_tag_1" {
      - id         = "Z2lkOi8vb3BzbGV2ZWwvVGVhbS8xNzQzNA:Z2lkOi8vb3BzbGV2ZWwvVGFnLzgyNjU0OTMw" -> null
      - key        = "hellothere" -> null
      - team       = "Z2lkOi8vb3BzbGV2ZWwvVGVhbS8xNzQzNA" -> null
      - team_alias = "team_platform_3_a" -> null
      - value      = "worldooooooooooooooo" -> null
    }

Plan: 0 to add, 0 to change, 1 to destroy.
opslevel_team_tag.team_tag_1: Destroying... [id=Z2lkOi8vb3BzbGV2ZWwvVGVhbS8xNzQzNA:Z2lkOi8vb3BzbGV2ZWwvVGFnLzgyNjU0OTMw]
opslevel_team_tag.team_tag_1: Destruction complete after 0s

Destroy complete! Resources: 1 destroyed.
```

### Re-create `opslevel_team_tag`, `terraform apply`
```tf
Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # opslevel_team_tag.team_tag_1 will be created
  + resource "opslevel_team_tag" "team_tag_1" {
      + id         = (known after apply)
      + key        = "key2"
      + team       = "Z2lkOi8vb3BzbGV2ZWwvVGVhbS8xNzQzNA"
      + team_alias = "team_platform_3_a"
      + value      = "value"
    }

Plan: 1 to add, 0 to change, 0 to destroy.
opslevel_team_tag.team_tag_1: Creating...
opslevel_team_tag.team_tag_1: Creation complete after 1s [id=Z2lkOi8vb3BzbGV2ZWwvVGVhbS8xNzQzNA:Z2lkOi8vb3BzbGV2ZWwvVGFnLzg5MzcwNzA1]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.
```

### update `value` field then `terraform apply`
```tf
opslevel_team_tag.team_tag_1: Refreshing state... [id=Z2lkOi8vb3BzbGV2ZWwvVGVhbS8xNzQzNA:Z2lkOi8vb3BzbGV2ZWwvVGFnLzg5MzcwNzA1]

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # opslevel_team_tag.team_tag_1 will be updated in-place
  ~ resource "opslevel_team_tag" "team_tag_1" {
        id         = "Z2lkOi8vb3BzbGV2ZWwvVGVhbS8xNzQzNA:Z2lkOi8vb3BzbGV2ZWwvVGFnLzg5MzcwNzA1"
      ~ value      = "value" -> "value123"
        # (3 unchanged attributes hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
opslevel_team_tag.team_tag_1: Modifying... [id=Z2lkOi8vb3BzbGV2ZWwvVGVhbS8xNzQzNA:Z2lkOi8vb3BzbGV2ZWwvVGFnLzg5MzcwNzA1]
opslevel_team_tag.team_tag_1: Modifications complete after 0s [id=Z2lkOi8vb3BzbGV2ZWwvVGVhbS8xNzQzNA:Z2lkOi8vb3BzbGV2ZWwvVGFnLzg5MzcwNzA1]

Apply complete! Resources: 0 added, 1 changed, 0 destroyed.
```

### Remove `team_tag` from Terraform state
```bash
> terraform state rm opslevel_team_tag.team_tag_1
Removed opslevel_team_tag.team_tag_1
Successfully removed 1 resource instance(s).
```

### Import `team_tag` with new tag ID
```bash
terraform import opslevel_team_tag.team_tag_1 'Z2lkOi8vb3BzbGV2ZWwvVGVhbS8xNzQzNA:Z2lkOi8vb3BzbGV2ZWwvVGFnLzg5MzcwNzUx'
```
outputs
```tf
opslevel_team_tag.team_tag_1: Importing from ID "Z2lkOi8vb3BzbGV2ZWwvVGVhbS8xNzQzNA:Z2lkOi8vb3BzbGV2ZWwvVGFnLzg5MzcwNzUx"...
opslevel_team_tag.team_tag_1: Import prepared!
  Prepared opslevel_team_tag for import
opslevel_team_tag.team_tag_1: Refreshing state... [id=Z2lkOi8vb3BzbGV2ZWwvVGVhbS8xNzQzNA:Z2lkOi8vb3BzbGV2ZWwvVGFnLzg5MzcwNzUx]

Import successful!

The resources that were imported are shown above. These resources are now in
your Terraform state and will henceforth be managed by Terraform.
```

### CODE REVIEW

File: opslevel/validators.go [hunk 1]
```diff
@@ -37,6 +37,40 @@ func IdStringValidator() validator.String {
 	return idStringValidator{}
 }
 
+// Tag - "<resource-id>:<tag-id>" String Validator
```
Comment (davidbloss) [inline opslevel/validators.go:40]: This is currently unused but doesn't hurt to have

File: opslevel/resource_opslevel_team_tag.go [hunk 2]
```diff
@@ -63,7 +65,7 @@ func (teamTagResource *TeamTagResource) Schema(ctx context.Context, req resource
 		MarkdownDescription: "Team Tag Resource",
 		Attributes: map[string]schema.Attribute{
 			"id": schema.StringAttribute{
-				Description: "The ID of this resource.",
+				Description: "The ID of this resource. Formatted as <team-id>:<tag-id>.",
```
Comment (rocktavious) [inline opslevel/resource_opslevel_team_tag.go:68]: I'm confused why we have to change this?  Can the `ImportState` function not do the splitting logic and then find the tag by ID and store this as the ID without this format?

Additionally `ImportState` should beable to do all the stuff it needs to.  I think you are conflating the `ImportStatePassthroughID` - we have to write our own importer function and not rely upon reusing the `Read` function.

**Review by rocktavious — CHANGES_REQUESTED** [review]
I don't think this is the correct change to make.  We need a custom `ImportState` and not use the `ImportStatePassthroughID`

**Comment by davidbloss** [comment]
Replaced by [this PR](https://github.com/OpsLevel/terraform-provider-opslevel/pull/366)

**Pull request closed without merging**